File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        Stack<String> stk = new Stack<>();
        
        for(int i=0; i<s.length(); i++) {
            String str = String.valueOf(s.charAt(i));
            if(str.equals("(")){
                stk.push(str);
            }
            else {
                if(stk.isEmpty())
                    return false;
                else {
                    stk.pop();
                }
            }
        }
        if(stk.size()!=0)
            return false;
        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;

    let stack = [];
    for(let x of s) {
        if(x=="(") {
            stack.push("(");
        }
        else if(x==")"&&stack.length!=0){
            stack.pop();
        }
        else if(x==")"&&stack.length==0) {
            return false;
        }
    }
    if(stack.length!=0) {
        return false;
    }

    return true;
}